### PR TITLE
Respect disabled nested queries on the frontend

### PR DESCRIPTION
**Pull request by kulyk**
Admins can completely disable nested queries on `/admin/settings/general`. This PR fixes the FE to respect that setting:

* hides the "Explore results" button for native questions
* hides "Saved Questions" option from the data selector

Related issue: #19341 (we should also reject nested queries execution on the BE if they're disabled, so not using the closing keywords here)

⚠️ Not backporting it automatically, as it will most likely fail due to new datasets code in the data selector. Will do it manually once this PR is merged

### To Verify

**Disable nested queries**
1. As an admin, go to `/admin/settings/general`
2. Scroll to the bottom and disable the "Enabled Nested Queries" option

**Native Questions**
1. Open any saved native question
2. Ensure the "Explore results" button is not shown at the top right

**QB Data Selector**
1. Ask a question > Simple / Custom Question
2. Ensure "Saved Questions" DB is removed from the list
3. Try with and without datasets in your instance, as the picker behaves differently in that case. When there are saved datasets, but nested queries are disabled, you shouldn't be able to pick a dataset either (because it's a saved question too)

### Demo

<details>
<summary>Native Question's "Explore results" button</summary>

**Before**
Even when nested queries are disabled, the "Explore results" button is still visible

enabled-native-question

**After**
Now when nested queries are disabled, the "Explore results" is hidden

disabled-native-question

</details>

<details>
<summary>QB Data Selector</summary>

**Before**
Even when nested queries are disabled, people can still pick "Datasets" or "Saved Questions"

CleanShot 2021-12-14 at 16 11 11@2x

**After**
Even when nested queries are disabled, Metabase only allows picking a raw table

CleanShot 2021-12-14 at 16 10 58@2x

</details>

### CODE REVIEW

File: src/metabase/public_settings.clj [hunk 1]
```diff
@@ -175,7 +175,8 @@
 (defsetting enable-nested-queries
   (deferred-tru "Allow using a saved question as the source for other queries?")
   :type    :boolean
-  :default true)
+  :default true
+  :visibility :authenticated)
```
Comment (kulyk) [inline src/metabase/public_settings.clj:179]: The only backend change here is changing the `enable-nested-queries` setting's visibility from `:admin` to `:authenticated`. This is needed so the FE can take in into account when a non-admin user is signed in

**Review by flamber — APPROVED** [review]
The frontend works as expected :+1: 
Is the search handled in #19380, since the code doesn't look it touches search?
image

Should we perhaps not backport this? It's a P2.5 issue in my book and given that it might break something for some (until the admin enable the setting again).

**Review by dpsutton — APPROVED** [review]
backend change is simple and fine 👍

**Comment by kulyk** [comment]
> Is the search handled in #19380, since the code doesn't look it touches search?

Nope, great catch! Extended the repro test and fixed it in the recent commit. Thanks @flamber 

> Should we perhaps not backport this? It's a P2.5 issue in my book and given that it might break something for some (until the admin enable the setting again).

SGTM 👍

**Review by daltojohnso — APPROVED** [review]

**Pull request merged**